web/impact/impact/v1/classes/option_analysis.py

```diff
@@ -24,38 +25,81 @@
 from impact.v1.helpers.matching_program_criterion_helper import (
     MatchingProgramCriterionHelper,
 )
+from accelerator_abstract.models.base_core_profile import (
+    UI_GENDER_CHOICES
+)
 
-
-CriterionHelper.register_helper(JudgeGenderCriterionHelper,
-                                "judge", "gender")
-CriterionHelper.register_helper(JudgeRoleCriterionHelper,
-                                "judge", "role")
-CriterionHelper.register_helper(MatchingIndustryCriterionHelper,
-                                "matching", "industry")
-CriterionHelper.register_helper(MatchingProgramCriterionHelper,
-                                "matching", "program")
+CriterionHelper.register_helper(
+    JudgeGenderCriterionHelper, "judge", "gender")
+CriterionHelper.register_helper(
+    JudgeRoleCriterionHelper, "judge", "role")
+CriterionHelper.register_helper(
+    MatchingIndustryCriterionHelper, "matching", "industry")
+CriterionHelper.register_helper(
+    MatchingProgramCriterionHelper, "matching", "program")
 
 
 class OptionAnalysis(object):
     _judge_to_count = None
+    expert_profile = "judge__expertprofile__"
+    industry_judge_field = expert_profile + "primary_industry__name"
+    program_judge_field = expert_profile + "home_program_family__name"
+    gender_judge_field = expert_profile + "gender"
+    role_judge_field = expert_profile + "expert_category__name"
 
     def __init__(self,
-                 option_spec,
                  apps,
                  app_ids,
                  judging_round,
                  application_counts,
                  criterion_helpers):
-        self.option_spec = option_spec
+        self.criterion_helpers = criterion_helpers
+        self.option_spec = None
         self.judging_round = judging_round
-        self.helper = CriterionOptionSpecHelper(
-            option_spec,
-            criterion_helpers)
+        self.cycle = self.judging_round.program.cycle
+        self.helper = None
         self.apps = apps
         self.app_ids = app_ids
         self.application_counts = application_counts
+        self.criterion_total_capacities = {}
+        self.judge_to_capacity_cache = None
+        self.criterion_total_functions = {
+            ("judge", "gender"): {
```

Comment:
Information specific to criteria and options should live on the appropriate helpers.